core/unix/module_elf.c

```diff
@@ -1085,6 +1085,9 @@ module_lookup_symbol(ELF_SYM_TYPE *sym, os_privmod_data_t *pd)
      */
     ASSERT_OWN_RECURSIVE_LOCK(true, &privload_lock);
     mod = privload_first_module();
+    /* Symbols are currently looked up following the dependency chain depth-first instead
+     * of breadth-first as reflected in the global list 'modlist' (xref i#3850).
```

Comment:
I found the "...as reflected" confusing: sounds like the global list is already breadth-first.   Please reword.

Reply:
s/as reflected in the global list 'modlist'//
I think the code is self-explaining, actually.